proto_parser/src/proto_parser.cc

```diff
@@ -0,0 +1,249 @@
+/*
+*   Copyright 2020 Google LLC
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*    https://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#include "proto_parser.h"
+
+namespace wireless_android_play_analytics {
+
+void ProtoParser::PopulateFields(int& prev_field_line, 
+  const google::protobuf::TextFormat::ParseInfoTree* tree,
+  const google::protobuf::Message& message,
+  std::shared_ptr<ProtoValue>& proto_value) {
```

Comment:
I think this function can be defined like

MessageValue GetMessageValue(const ParseInfoTree& tree, const Message& message, int& prev_filed_line)